### PR TITLE
conf: Parse boolean values case insensitively

### LINKED ISSUE #155: Boolean values should be accepted even if upper-cased
Only the lower-cased variant are considered for boolean conversion:
  https://github.com/dalibo/pgtoolkit/blob/5c45afe24123a1813a62b8abe2759fca83fdc0ae/pgtoolkit/conf.py#L260
  
While the PG doc states it should be case insensitive: 
  https://www.postgresql.org/docs/current/config-setting.html#CONFIG-SETTING-NAMES-VALUES

### PR DESCRIPTION
**Pull request by pgiraud**
Fixes #155

### CODE REVIEW

**Review by l00ptr — APPROVED** [review]
LGTM, but how do we handle the changelog ?

**Comment by pgiraud** [comment]
> LGTM, but how do we handle the changelog ?

That's a good question. Currently, we don't have any changelog currently. Also, this change is not a breaking one.

File: tests/test_conf.py [hunk 1]
```diff
@@ -9,13 +10,20 @@
...
+    for v, r in [
+        ("on", True),
+        ("off", False),
+        ("true", True),
+        ("false", False),
+        ("yes", True),
+        ("'no'", False),
+    ]:
+        assert parse_value(v) is r
+        assert parse_value(random_upper(v)) is r
```
Comment (dlax) [inline tests/test_conf.py:26]: Not sure we need this extra complexity:
- in my experience loops in tests are hard to debug (might be quite easy in this case, but in general it's not); and that's why parametrized tests exist (by the way, it'd be nice to refactor this test to make it parametrized, but that's another issue)
- the fact that the randomized value is not readily available (one would need to change this code through an extra variable to see it) is also problematic
- so in short, I think we could keep the previous logic and just change e.g. `true` into `TRUE`, `no` into `No`; this would be largely enough for coverage

**Review by dlax — CHANGES_REQUESTED** [review]
As a follow-up, we could also handle the "any unambiguous prefix of one of these" that's mentioned in https://www.postgresql.org/docs/current/config-setting.html#CONFIG-SETTING-NAMES-VALUES

**Comment by pgiraud** [comment]
> As a follow-up, we could also handle the "any unambiguous prefix of one of these" that's mentioned in https://www.postgresql.org/docs/current/config-setting.html#CONFIG-SETTING-NAMES-VALUES

I saw this in the documentation and to be sincere, I didn't understand what it means.

**Comment by dlax** [comment]
> > As a follow-up, we could also handle the "any unambiguous prefix of one of these" that's mentioned in https://www.postgresql.org/docs/current/config-setting.html#CONFIG-SETTING-NAMES-VALUES
> 
> I saw this in the documentation and to be sincere, I didn't understand what it means.

It means that `tr` is understood as `True`, but `o` is not accepted for either `True` or `False` because it's an ambiguous prefix of `on` and `off`.

**Comment by pgiraud** [comment]
> It means that `tr` is understood as `True`, but `o` is not accepted for either `True` or `False` because it's an ambiguous prefix of `on` and `off`.

The problem is we can't be sure that `tr` is not a string. On the same manner, we assume that when `0`  should be parsed  to a number value even though it could be a valid boolean value. Same for `1`.

**Comment by dlax** [comment]
> > It means that `tr` is understood as `True`, but `o` is not accepted for either `True` or `False` because it's an ambiguous prefix of `on` and `off`.
> 
> The problem is we can't be sure that `tr` is not a string. On the same manner, we assume that when `0` should be parsed to a number value even though it could be a valid boolean value. Same for `1`.

Maybe we'd need to see how postgresql handles such values? But as it knows the type of the setting, it's easier. Let's forget about this if it's not doable; at least the discussion was not useless.

**Review by dlax — APPROVED** [review]
